cmd/tk/workflow.go

```diff
@@ -25,6 +25,16 @@ func workflowFlags(fs *pflag.FlagSet) *workflowFlagVars {
 	return &v
 }
 
+func applyDeleteFlags(fs *pflag.FlagSet) kubernetes.ApplyDeleteOpts {
+	force := fs.Bool("force", false, "force operating (kubectl --force)")
+	autoApprove := fs.Bool("dangerous-auto-approve", false, "skip interactive approval. Only for automation!")
+
+	return kubernetes.ApplyDeleteOpts{
```

Comment:
This most probably won‘t work that way either, because you dereference the pointers here immediately, before `pflag.Parse` is able to update the values. 

Rather create a local `kubernetes.ApplyDeleteOpts` variable and use `fs.BoolVar(&opts.Force, ...)`, so that the underlying flag holds a pointer to the actual returned value and can update it accordingly

Reply:
test at local:
```
# ./tk apply --force
Error: accepts 1 arg(s), received 0
Usage:
  tk apply [path] [flags]

Flags:
      --dangerous-auto-approve   skip interactive approval. Only for automation!
      --force                    force applying (kubectl apply --force)
  -h, --help                     help for apply
  -t, --target strings           only use the specified objects (Format: <type>/<name>)

Global Flags:
  -v, --verbose
```
The flag has registered.

Reply:
Since i pass a pointer to `func applyDeleteFlags(fs *pflag.FlagSet) kubernetes.ApplyDeleteOpts`, i think it can be parsed.

Reply:
This is a bit tricky here, you don't actually pass a pointer:

The `flag` package creates new variables in memory, adds pointers to these into underlying [`flag.Flag`](https://golang.org/pkg/flag/#Flag) and returns those pointers to you as well:
https://github.com/grafana/tanka/blob/71e9f5ad849a6d18327be6166f05cddc1abd2f66/cmd/tk/workflow.go#L29-L30 

In case of a `flag.Parse`, the underlying values of `autoApprove` and `force` would be updated.

However, here:
https://github.com/grafana/tanka/blob/71e9f5ad849a6d18327be6166f05cddc1abd2f66/cmd/tk/workflow.go#L33-L34
you **dereference** the pointers, which means you make a **copy of the _current_ value**. This means that the `kubernetes.ApplyDeleteOpts` includes the concrete value at the time of running this code, which is *before `pflag.Parse`*

Once the flags are parsed, the underlying value of the flag is updated, but as you already dereferenced the pointers you do not receive this change.

Instead, use a `flagVar`:
> Rather create a local kubernetes.ApplyDeleteOpts variable and use fs.BoolVar(&opts.Force, ...), so that the underlying flag holds a pointer to the actual returned value and can update it accordingly

Reply:
learn it, thanks @sh0rez